### ISSUE TITLE
MDX story snippets

### ISSUE THREAD

**Issue by zeorin** [rbardini/storybook-addon-playroom#15]
I'm using the snippet generation in my project, and it works great. But I'd like to be able to have [MDX stories](https://github.com/storybookjs/storybook/blob/master/addons/docs/docs/mdx.md) be generated as snippets, too.

**Comment by rbardini** [comment]
Hi @zeorin! I never tried it with MDX, but it might work if we add `mdx` to the supported extensions:

https://github.com/rbardini/storybook-addon-playroom/blob/a0b23ca40e526c7654933e3a4c66afbfa7936f6f/src/generateSnippets.tsx#L40

You can give it a try by manually changing the file in `node_modules`. We can incorporate the change into the source code if it works, otherwise it may require more investigation.

**Comment by zeorin** [comment]
I tried that already, actually:
```
Unexpected error: SyntaxError: Cannot use import statement outside a module
```

**Comment by umar-ahmed** [comment]
Seems like it might require more investigation to try and support mdx based stories. In the meantime, is there any way to filter out the stories that use `mdx`? Right now it seems like if you have a mix of `mdx` and `js` based stories the snippet generator will fail with the complaint @zeorin mentioned.

**Comment by umar-ahmed** [comment]
I'm pretty sure the `generateSnippets` script is based on the one [here in the storyshots addon](https://github.com/storybookjs/storybook/blob/next/addons/storyshots/storyshots-core/src/frameworks/configure.ts).

One thing I noticed is that they mention that in order for that code to work with MDX based stories, you have to apply this transform that they provide for Jest: https://github.com/storybookjs/storybook/blob/next/addons/storyshots/storyshots-core/README.md#configure-jest-for-mdx-docs-add-on-stories.

It basically boils down to running their MDX compiler plugin like so:

https://github.com/storybookjs/storybook/blob/8ab8425ec556f7350bcc211154ed26efd62a7508/addons/docs/jest-transform-mdx.js#L12-L22

This makes sense in a Jest environment, but I'm not quite sure how you would apply that same idea here

**Comment by rbardini** [comment]
Indeed, the script is based on StoryShots addon, with Jest dependencies removed. I'm not sure we can do the same here though, as the Storybook client API doesn't seem to support transformers—if it had, the Jest setup would not be needed in the first place.

Haven't had time to look into this yet, but if anyone wants to give it a try, a PR would be very welcome 🙂

**Comment by umar-ahmed** [comment]
@rbardini I can give it a go.

The approach I'm going try is described here: https://github.com/mdx-js/mdx/issues/171#issuecomment-409792001. I'm going to hijack Node's `require` and try to use something similar to the Jest transform linked earlier.

If someone thinks of a better solution, please do share! :smile:

**Comment by umar-ahmed** [comment]
Ran out of time today, but I managed to get something kinda working: https://github.com/umar-ahmed/storybook-addon-playroom.

The main issue I am seeing is that for React elements, it converts it to `MDXCreateElement`:

![image](https://user-images.githubusercontent.com/8302959/87234784-8e6f5280-c3a2-11ea-9f6e-4f9ac6d2343e.png)

**Comment by rbardini** [comment]
That looks very promising, @umar-ahmed! Nice work 👍

Don't know how to get rid of the `MDXCreateElement` elements, but if nothing else works we might use the `displayName` and `filterProps` [options from react-element-to-jsx-string](https://github.com/algolia/react-element-to-jsx-string#reactelementtojsxstringreactelement-options).
